pkg/jerryscript/Makefile.jerryscript

```diff
@@ -1,15 +1,14 @@
-BUILD_DIR  ?= $(CURDIR)/riot
+BUILD_DIR  ?= $(BINDIR)/jerryscript
```

Comment:
why not use `BUILD_DIR  ?= $(PKG_BUILDDIR)` instead?

Reply:
First it is not defined there, and also it is the directory where the package is cloned.

As described in the commit message and the title of this PR. It is the goal on this one to move toward out of source build, and all normal modules in RIOT build in `$(BINDIR)/module_name`.

With out of source build, the package could be cloned only once for all platforms (20M * 120 boards == a lot).

Reply:
ok, then we definitely need to work on the naming of RIOT directories. Because if `PKG_BUILDDIR` is the directory where the package is cloned and *not* build, its very confusing - the name is why I suggested to use that for `BUILD_DIR`.

Reply:
Some of the packages are still build into there when they use their own build system.